File: output.java
package proxy.java.util;
public class Random extends java.util.Random
{
    private Random_callbackInterface callbackInterface = null;
    public  Random(Random_callbackInterface callbackInterface)
    {
        super();
        this.callbackInterface = callbackInterface;
    }
    public  Random(Random_callbackInterface callbackInterface, long arg0)
    {
        super(arg0);
        this.callbackInterface = callbackInterface;
    }
    public void _python_setSeed(long arg0)
    {
        super.setSeed(arg0);
    }
    public void setSeed(long arg0)
    {
        if(this.callbackInterface != null) { this.callbackInterface.setSeed(arg0); }
        else { super.setSeed(arg0); }
    }
    public void _python_nextBytes(byte[] arg0)
    {
        super.nextBytes(arg0);
    }
    public void nextBytes(byte[] arg0)
    {
        if(this.callbackInterface != null) { this.callbackInterface.nextBytes(arg0); }
        else { super.nextBytes(arg0); }
    }
    public long _python_nextLong()
    {
        return super.nextLong();
    }
    public long nextLong()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.nextLong(); }
        else { return super.nextLong(); }
    }
    public boolean _python_nextBoolean()
    {
        return super.nextBoolean();
    }
    public boolean nextBoolean()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.nextBoolean(); }
        else { return super.nextBoolean(); }
    }
    public float _python_nextFloat()
    {
        return super.nextFloat();
    }
    public float nextFloat()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.nextFloat(); }
        else { return super.nextFloat(); }
    }
    public double _python_nextGaussian()
    {
        return super.nextGaussian();
    }
    public double nextGaussian()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.nextGaussian(); }
        else { return super.nextGaussian(); }
    }
    public java.util.stream.IntStream _python_ints()
    {
        return super.ints();
    }
    public java.util.stream.IntStream ints()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.ints(); }
        else { return super.ints(); }
    }
    public java.util.stream.IntStream _python_ints(long arg0, int arg1, int arg2)
    {
        return super.ints(arg0, arg1, arg2);
    }
    public java.util.stream.IntStream ints(long arg0, int arg1, int arg2)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.ints(arg0, arg1, arg2); }
        else { return super.ints(arg0, arg1, arg2); }
    }
    public java.util.stream.IntStream _python_ints(int arg0, int arg1)
    {
        return super.ints(arg0, arg1);
    }
    public java.util.stream.IntStream ints(int arg0, int arg1)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.ints(arg0, arg1); }
        else { return super.ints(arg0, arg1); }
    }
    public java.util.stream.IntStream _python_ints(long arg0)
    {
        return super.ints(arg0);
    }
    public java.util.stream.IntStream ints(long arg0)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.ints(arg0); }
        else { return super.ints(arg0); }
    }
    public java.util.stream.LongStream _python_longs(long arg0, long arg1, long arg2)
    {
        return super.longs(arg0, arg1, arg2);
    }
    public java.util.stream.LongStream longs(long arg0, long arg1, long arg2)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.longs(arg0, arg1, arg2); }
        else { return super.longs(arg0, arg1, arg2); }
    }
    public java.util.stream.LongStream _python_longs(long arg0)
    {
        return super.longs(arg0);
    }
    public java.util.stream.LongStream longs(long arg0)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.longs(arg0); }
        else { return super.longs(arg0); }
    }
    public java.util.stream.LongStream _python_longs()
    {
        return super.longs();
    }
    public java.util.stream.LongStream longs()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.longs(); }
        else { return super.longs(); }
    }
    public java.util.stream.LongStream _python_longs(long arg0, long arg1)
    {
        return super.longs(arg0, arg1);
    }
    public java.util.stream.LongStream longs(long arg0, long arg1)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.longs(arg0, arg1); }
        else { return super.longs(arg0, arg1); }
    }
    public java.util.stream.DoubleStream _python_doubles(double arg0, double arg1)
    {
        return super.doubles(arg0, arg1);
    }
    public java.util.stream.DoubleStream doubles(double arg0, double arg1)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.doubles(arg0, arg1); }
        else { return super.doubles(arg0, arg1); }
    }
    public java.util.stream.DoubleStream _python_doubles(long arg0, double arg1, double arg2)
    {
        return super.doubles(arg0, arg1, arg2);
    }
    public java.util.stream.DoubleStream doubles(long arg0, double arg1, double arg2)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.doubles(arg0, arg1, arg2); }
        else { return super.doubles(arg0, arg1, arg2); }
    }
    public java.util.stream.DoubleStream _python_doubles()
    {
        return super.doubles();
    }
    public java.util.stream.DoubleStream doubles()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.doubles(); }
        else { return super.doubles(); }
    }
    public java.util.stream.DoubleStream _python_doubles(long arg0)
    {
        return super.doubles(arg0);
    }
    public java.util.stream.DoubleStream doubles(long arg0)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.doubles(arg0); }
        else { return super.doubles(arg0); }
    }
    public int _python_nextInt(int arg0)
    {
        return super.nextInt(arg0);
    }
    public int nextInt(int arg0)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.nextInt(arg0); }
        else { return super.nextInt(arg0); }
    }
    public int _python_nextInt()
    {
        return super.nextInt();
    }
    public int nextInt()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.nextInt(); }
        else { return super.nextInt(); }
    }
    public double _python_nextDouble()
    {
        return super.nextDouble();
    }
    public double nextDouble()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.nextDouble(); }
        else { return super.nextDouble(); }
    }
    public void _python_wait(long arg0, int arg1) throws java.lang.InterruptedException
    {
        super.wait(arg0, arg1);
    }
    public void _python_wait(long arg0) throws java.lang.InterruptedException
    {
        super.wait(arg0);
    }
    public void _python_wait() throws java.lang.InterruptedException
    {
        super.wait();
    }
    public boolean _python_equals(java.lang.Object arg0)
    {
        return super.equals(arg0);
    }
    public boolean equals(java.lang.Object arg0)
    {
        if(this.callbackInterface != null) { return this.callbackInterface.equals(arg0); }
        else { return super.equals(arg0); }
    }
    public java.lang.String _python_toString()
    {
        return super.toString();
    }
    public java.lang.String toString()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.toString(); }
        else { return super.toString(); }
    }
    public int _python_hashCode()
    {
        return super.hashCode();
    }
    public int hashCode()
    {
        if(this.callbackInterface != null) { return this.callbackInterface.hashCode(); }
        else { return super.hashCode(); }
    }
    public java.lang.Class _python_getClass()
    {
        return super.getClass();
    }
    public void _python_notify()
    {
        super.notify();
    }
    public void _python_notifyAll()
    {
        super.notifyAll();
    }
}
package proxy.java.util;
public interface Random_callbackInterface
{
    public void setSeed(long arg0) ;
    public void nextBytes(byte[] arg0) ;
    public long nextLong() ;
    public boolean nextBoolean() ;
    public float nextFloat() ;
    public double nextGaussian() ;
    public java.util.stream.IntStream ints() ;
    public java.util.stream.IntStream ints(long arg0, int arg1, int arg2) ;
    public java.util.stream.IntStream ints(int arg0, int arg1) ;
    public java.util.stream.IntStream ints(long arg0) ;
    public java.util.stream.LongStream longs(long arg0, long arg1, long arg2) ;
    public java.util.stream.LongStream longs(long arg0) ;
    public java.util.stream.LongStream longs() ;
    public java.util.stream.LongStream longs(long arg0, long arg1) ;
    public java.util.stream.DoubleStream doubles(double arg0, double arg1) ;
    public java.util.stream.DoubleStream doubles(long arg0, double arg1, double arg2) ;
    public java.util.stream.DoubleStream doubles() ;
    public java.util.stream.DoubleStream doubles(long arg0) ;
    public int nextInt(int arg0) ;
    public int nextInt() ;
    public double nextDouble() ;
    public boolean equals(java.lang.Object arg0) ;
    public java.lang.String toString() ;
    public int hashCode() ;
}

File: output.py
from jnius import autoclass, PythonJavaClass, java_method, JavaClass, MetaJavaClass
class Random:
    def __init__(self, *args):
        super(Random, self).__init__(*args)
        self.override_callback = _RandomOverrideCallback(self)
        
        self._proxy = autoclass("proxy.java.util.Random")(self.override_callback, *args)
        
         
    def setSeed(self, *args):
        self._proxy._python_setSeed(*args)
         
    def nextBytes(self, *args):
        self._proxy._python_nextBytes(*args)
         
    def nextLong(self, *args):
        return self._proxy._python_nextLong(*args)
         
    def nextBoolean(self, *args):
        return self._proxy._python_nextBoolean(*args)
         
    def nextFloat(self, *args):
        return self._proxy._python_nextFloat(*args)
         
    def nextGaussian(self, *args):
        return self._proxy._python_nextGaussian(*args)
         
    def ints(self, *args):
        return self._proxy._python_ints(*args)
         
    def ints(self, *args):
        return self._proxy._python_ints(*args)
         
    def ints(self, *args):
        return self._proxy._python_ints(*args)
         
    def ints(self, *args):
        return self._proxy._python_ints(*args)
         
    def longs(self, *args):
        return self._proxy._python_longs(*args)
         
    def longs(self, *args):
        return self._proxy._python_longs(*args)
         
    def longs(self, *args):
        return self._proxy._python_longs(*args)
         
    def longs(self, *args):
        return self._proxy._python_longs(*args)
         
    def doubles(self, *args):
        return self._proxy._python_doubles(*args)
         
    def doubles(self, *args):
        return self._proxy._python_doubles(*args)
         
    def doubles(self, *args):
        return self._proxy._python_doubles(*args)
         
    def doubles(self, *args):
        return self._proxy._python_doubles(*args)
         
    def nextInt(self, *args):
        return self._proxy._python_nextInt(*args)
         
    def nextInt(self, *args):
        return self._proxy._python_nextInt(*args)
         
    def nextDouble(self, *args):
        return self._proxy._python_nextDouble(*args)
         
    def wait(self, *args):
        self._proxy._python_wait(*args)
         
    def wait(self, *args):
        self._proxy._python_wait(*args)
         
    def wait(self, *args):
        self._proxy._python_wait(*args)
         
    def equals(self, *args):
        return self._proxy._python_equals(*args)
         
    def toString(self, *args):
        return self._proxy._python_toString(*args)
         
    def hashCode(self, *args):
        return self._proxy._python_hashCode(*args)
         
    def getClass(self, *args):
        return self._proxy._python_getClass(*args)
         
    def notify(self, *args):
        self._proxy._python_notify(*args)
         
    def notifyAll(self, *args):
        self._proxy._python_notifyAll(*args)
         
from jnius import autoclass, PythonJavaClass, java_method, JavaClass, MetaJavaClass
class _RandomOverrideCallback(PythonJavaClass):
    __javainterfaces__ = ['proxy/java/util/Random_callbackInterface']
    def __init__(self, proxy):
        super(_RandomOverrideCallback, self).__init__()
        self._proxy = proxy
         
    @java_method("(J)V", "setSeed")
    def setSeed0(self, *args):
        self._proxy.setSeed(*args)
         
    @java_method("(Lbyte[];)V", "nextBytes")
    def nextBytes1(self, *args):
        self._proxy.nextBytes(*args)
         
    @java_method("()J", "nextLong")
    def nextLong2(self, *args):
        self._proxy.nextLong(*args)
         
    @java_method("()Z", "nextBoolean")
    def nextBoolean3(self, *args):
        self._proxy.nextBoolean(*args)
         
    @java_method("()F", "nextFloat")
    def nextFloat4(self, *args):
        self._proxy.nextFloat(*args)
         
    @java_method("()D", "nextGaussian")
    def nextGaussian5(self, *args):
        self._proxy.nextGaussian(*args)
         
    @java_method("()Ljava/util/stream/IntStream;", "ints")
    def ints6(self, *args):
        self._proxy.ints(*args)
         
    @java_method("(JII)Ljava/util/stream/IntStream;", "ints")
    def ints7(self, *args):
        self._proxy.ints(*args)
         
    @java_method("(II)Ljava/util/stream/IntStream;", "ints")
    def ints8(self, *args):
        self._proxy.ints(*args)
         
    @java_method("(J)Ljava/util/stream/IntStream;", "ints")
    def ints9(self, *args):
        self._proxy.ints(*args)
         
    @java_method("(JJJ)Ljava/util/stream/LongStream;", "longs")
    def longs10(self, *args):
        self._proxy.longs(*args)
         
    @java_method("(J)Ljava/util/stream/LongStream;", "longs")
    def longs11(self, *args):
        self._proxy.longs(*args)
         
    @java_method("()Ljava/util/stream/LongStream;", "longs")
    def longs12(self, *args):
        self._proxy.longs(*args)
         
    @java_method("(JJ)Ljava/util/stream/LongStream;", "longs")
    def longs13(self, *args):
        self._proxy.longs(*args)
         
    @java_method("(DD)Ljava/util/stream/DoubleStream;", "doubles")
    def doubles14(self, *args):
        self._proxy.doubles(*args)
         
    @java_method("(JDD)Ljava/util/stream/DoubleStream;", "doubles")
    def doubles15(self, *args):
        self._proxy.doubles(*args)
         
    @java_method("()Ljava/util/stream/DoubleStream;", "doubles")
    def doubles16(self, *args):
        self._proxy.doubles(*args)
         
    @java_method("(J)Ljava/util/stream/DoubleStream;", "doubles")
    def doubles17(self, *args):
        self._proxy.doubles(*args)
         
    @java_method("(I)I", "nextInt")
    def nextInt18(self, *args):
        self._proxy.nextInt(*args)
         
    @java_method("()I", "nextInt")
    def nextInt19(self, *args):
        self._proxy.nextInt(*args)
         
    @java_method("()D", "nextDouble")
    def nextDouble20(self, *args):
        self._proxy.nextDouble(*args)
         
    @java_method("(JI)V", "wait")
    def wait21(self, *args):
        self._proxy.wait(*args)
         
    @java_method("(J)V", "wait")
    def wait22(self, *args):
        self._proxy.wait(*args)
         
    @java_method("()V", "wait")
    def wait23(self, *args):
        self._proxy.wait(*args)
         
    @java_method("(Ljava/lang/Object;)Z", "equals")
    def equals24(self, *args):
        self._proxy.equals(*args)
         
    @java_method("()Ljava/lang/String;", "toString")
    def toString25(self, *args):
        self._proxy.toString(*args)
         
    @java_method("()I", "hashCode")
    def hashCode26(self, *args):
        self._proxy.hashCode(*args)
         
    @java_method("()Ljava/lang/Class;", "getClass")
    def getClass27(self, *args):
        self._proxy.getClass(*args)
         
    @java_method("()V", "notify")
    def notify28(self, *args):
        self._proxy.notify(*args)
         
    @java_method("()V", "notifyAll")
    def notifyAll29(self, *args):
        self._proxy.notifyAll(*args)
         

